### PR TITLE
Add MinHook to third_party

### PR DESCRIPTION
**Pull request by pierricgimmig**
"The Minimalistic x86/x64 API Hooking Library for Windows"
This is a snapshot of https://github.com/TsudaKageyu/minhook at commit
4a455528f61b5a375b1f9d44e7d296d47f18bb18.

### CODE REVIEW

**Comment by pierricgimmig** [comment]
Hi @beckerhe , please let me know if there any special consideration regarding adding this to the internal repo. Thanks!

File: third_party/minhook/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    minhook-targets
+  NAMESPACE 
+    minhook::
+  DESTINATION 
+    "lib/minhook"
+)
+
+install(
+  DIRECTORY include DESTINATION .
+)
```
Comment (beckerhe) [inline third_party/minhook/CMakeLists.txt:141]: This part should be remove. It might otherwise affect our packaging process.

Comment (beckerhe) [reply third_party/minhook/CMakeLists.txt:141]: If you prefer to not make any changes to this upstream file we could alternatively write a FindMinhook.cmake script and put it into our `cmake/` subdirectory. Then this file wouldn't be used and we would preserve the upstream state. Might make updates easier.

**Review by beckerhe — APPROVED** [review]
Othewise LGTM. From my side the only thing missing is the license review (which shouldn't be a big deal since it's all BSD-2)

**Comment by beckerhe** [comment]
> Hi @beckerhe , please let me know if there any special consideration regarding adding this to the internal repo. Thanks!

No worries. Since that's Windows-only it's currently not relevant for the internal build.

**Review by danielfenner — APPROVED** [review]
lgtm

File: third_party/minhook/CMakeLists.txt (hunk 1 above)
Comment (pierricgimmig) [reply third_party/minhook/CMakeLists.txt]: Ah yes good point, I commented out the section and left a comment. Thanks for catching this.

**Pull request merged**